.github/workflows/helm.yml

```diff
@@ -0,0 +1,82 @@
+name: Helm Chart Tests
+
+on:
+  pull_request:
+  workflow_dispatch:
+
+jobs:
+  test-build:
+    name: test-helm-chart
+    runs-on: ubuntu-latest
+    env:
+      IMG: skyscanner/kms-issuer:dev
+      CERT_MANAGER_VERSION: v1.8.0
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3.0.2
+        with:
+          fetch-depth: 0
+
+      - uses: actions/setup-go@v3
+        with:
+          go-version: 1.17
+
+      - name: Set up Helm
+        uses: azure/setup-helm@v2.1
+        with:
+          version: v3.5.2
+
+      # Python is required because `ct lint` runs Yamale (https://github.com/23andMe/Yamale) and
+      # yamllint (https://github.com/adrienverge/yamllint) which require Python
+      - uses: actions/setup-python@v3.1.2
+        with:
+          python-version: '3.x'
+
+      - name: Set up chart-testing
+        uses: helm/chart-testing-action@v2.2.1
+        with:
+          version: v3.6.0
+
+      - name: Run chart-testing (list-changed)
+        id: list-changed
+        run: |
+          changed=$(ct list-changed --config ct.yaml)
+          if [[ -n "$changed" ]]; then
+            echo "::set-output name=changed::true"
+          fi
+
+      - name: Run chart-testing (lint)
+        run: ct lint --config ct.yaml
+        if: steps.list-changed.outputs.changed == 'true'
+
+      - name: Create k8s Kind Cluster
+        uses: helm/kind-action@v1.2.0
+        with:
+          cluster_name: kind
+        if: steps.list-changed.outputs.changed == 'true'
+
+      - name: Install cert-manager
+        run: kubectl apply -f https://github.com/cert-manager/cert-manager/releases/download/${CERT_MANAGER_VERSION}/cert-manager.yaml
+        if: steps.list-changed.outputs.changed == 'true'
+
+      - name: Build the testing kms-issuer docker image
+        run: docker build -t ${IMG} .
```

Comment:
we do a lot of building the docker image, creating kind clusters, etc. Might be nice if we could share some of this in a single workflow. Using caches between jobs?